### PR TITLE
Document node labels propagation feature

### LINKED ISSUE #5098: Document node labels propagation feature
Following up https://github.com/elastic/cloud-on-k8s/pull/5054 we should add some documentation and some examples.

### PR DESCRIPTION
**Pull request by barkbay**
Fix #5098

### CODE REVIEW

File: config/samples/elasticsearch/elasticsearch.yaml [hunk 1]
```diff
@@ -2,6 +2,9 @@
 apiVersion: elasticsearch.k8s.elastic.co/v1
 kind: Elasticsearch
 metadata:
+  #annotations:
+  #  uncomment the line below to copy the specified node labels as pod annotations and use it as an environment variable in the Pods
```
Comment (pebrc) [inline config/samples/elasticsearch/elasticsearch.yaml:6]: Nit: the user would also have to uncomment the line above :-)

File: docs/orchestrating-elastic-stack-applications/elasticsearch/advanced-node-scheduling.asciidoc [hunk 2]
```diff
@@ -200,57 +200,65 @@ spec:
...
+
+[id="{p}-availability-zone-awareness-downward-api"]
+=== Exposing Kubernetes node topology labels in Pods
+
+. First, ensure that the operator's flag `exposed-node-labels` contains the list of the Kubernetes node labels that should be exposed in the Elasticsearch Pods. If you are using the provided installation manifest, or the Helm chart, then this flag is already set with some well-known node labels that describe Kubernetes cluster topology, like `topology.kubernetes.io/.*` and `failure-domain.beta.kubernetes.io/.*`.
+. On the Elasticsearch resources set the `eck.k8s.elastic.co/downward-node-labels` annotations with the list of the Kubernetes node labels that must be copied as Pods annotations.
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/advanced-node-scheduling.asciidoc:211]:
```suggestion
. On the Elasticsearch resources set the `eck.k8s.elastic.co/downward-node-labels` annotations with the list of the Kubernetes node labels that should be copied as Pod annotations.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/advanced-node-scheduling.asciidoc [hunk 3]
```diff
@@ -200,57 +200,65 @@ spec:
...
+
+Refer to the next section or to the link:{eck_github}/tree/{eck_release_branch}/config/samples/elasticsearch/elasticsearch.yaml[Elasticsearch sample resource in the ECK source repository] for a complete example.
+
+[id="{p}-availability-zone-awareness-example"]
+=== Using node topology labels, Kubernetes topology spread constraints, and Elasticsearch shard allocation awareness
+
+The following example demonstrates how to use the `topology.kubernetes.io/zone` node labels to spread a NodeSet among the availability zones of a Kubernetes cluster.
+
+Note that by default ECK creates a `k8s_node_name` attribute with the name of the Kubernetes node running the Pod, and configures Elasticsearch to use this attribute. This ensures that Elasticsearch allocates primary and replica shards to pods running on different Kubernetes nodes and never to pods that are scheduled onto a single Kubernetes node. To preserve this behavior while making Elasticsearch aware of the availability zone, include the `k8s_node_name` attribute in the comma-separated `cluster.routing.allocation.awareness.attributes` list.
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/advanced-node-scheduling.asciidoc:221]:
```suggestion
Note that by default ECK creates a `k8s_node_name` attribute with the name of the Kubernetes node running the Pod, and configures Elasticsearch to use this attribute. This ensures that Elasticsearch allocates primary and replica shards to Pods running on different Kubernetes nodes and never to Pods that are scheduled onto the same Kubernetes node. To preserve this behavior while making Elasticsearch aware of the availability zone, include the `k8s_node_name` attribute in the comma-separated `cluster.routing.allocation.awareness.attributes` list.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/advanced-node-scheduling.asciidoc [hunk 4]
```diff
@@ -200,57 +200,65 @@ spec:
...
+. On the Elasticsearch resources set the `eck.k8s.elastic.co/downward-node-labels` annotations with the list of the Kubernetes node labels that must be copied as Pods annotations.
+. Use the link:https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/[Kubernetes downward API] in the `podTemplate` to make those annotations available as environment variables in Elasticsearch Pods.
+
+Refer to the next section or to the link:{eck_github}/tree/{eck_release_branch}/config/samples/elasticsearch/elasticsearch.yaml[Elasticsearch sample resource in the ECK source repository] for a complete example.
+
+[id="{p}-availability-zone-awareness-example"]
+=== Using node topology labels, Kubernetes topology spread constraints, and Elasticsearch shard allocation awareness
+
+The following example demonstrates how to use the `topology.kubernetes.io/zone` node labels to spread a NodeSet among the availability zones of a Kubernetes cluster.
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/advanced-node-scheduling.asciidoc:219]:
```suggestion
The following example demonstrates how to use the `topology.kubernetes.io/zone` node labels to spread a NodeSet across the availability zones of a Kubernetes cluster.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/advanced-node-scheduling.asciidoc [hunk 5]
```diff
@@ -200,57 +200,65 @@ spec:
...
+Starting with ECK 2.0 the operator can make Kubernetes Node labels available as Pod annotations. It can be used to make information, such as logical failure domains, available in a running Pod. Combined with link:https://www.elastic.co/guide/en/elasticsearch/reference/current/allocation-awareness.html#allocation-awareness[Elasticsearch shard allocation awareness] and link:https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/[Kubernetes topology spread constraints], you can create an availability zone-aware Elasticsearch cluster.
+
+[id="{p}-availability-zone-awareness-downward-api"]
+=== Exposing Kubernetes node topology labels in Pods
+
+. First, ensure that the operator's flag `exposed-node-labels` contains the list of the Kubernetes node labels that should be exposed in the Elasticsearch Pods. If you are using the provided installation manifest, or the Helm chart, then this flag is already set with some well-known node labels that describe Kubernetes cluster topology, like `topology.kubernetes.io/.*` and `failure-domain.beta.kubernetes.io/.*`.
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/advanced-node-scheduling.asciidoc:210]:
```suggestion
. First, ensure that the operator's flag `exposed-node-labels` contains the list of the Kubernetes node labels that should be exposed in the Elasticsearch Pods. If you are using the provided installation manifest, or the Helm chart, then this flag is already preset with two wildcard patterns for well-known node labels that describe Kubernetes cluster topology, `topology.kubernetes.io/.*` and `failure-domain.beta.kubernetes.io/.*`.
```
or something along those lines.

**Review by pebrc — APPROVED** [review]
LGTM

File: docs/operating-eck/operator-config.asciidoc [hunk 6]
```diff
@@ -26,6 +26,7 @@ ECK can be configured using either command line flags or environment variables.
 |enable-tracing | false | Enable APM tracing in the operator process. Use environment variables to configure APM server URL, credentials, and so on. See link:https://www.elastic.co/guide/en/apm/agent/go/1.x/configuration.html[Apm Go Agent reference] for details.
 |enable-webhook | false | Enables a validating webhook server in the operator process.
 |enforce-rbac-on-refs| false | Enables restrictions on cross-namespace resource association through RBAC.
+|exposed-node-labels|""| List of Kubernetes node labels which are allowed to be copied as annotations on the Elasticsearch Pods. See <<{p}-availability-zone-awareness>> for more details.
```
Comment (alaudazzi) [inline docs/operating-eck/operator-config.asciidoc:29]:
```suggestion
|exposed-node-labels|""| List of Kubernetes node labels which are allowed to be copied as annotations on the Elasticsearch Pods. Check <<{p}-availability-zone-awareness>> for more details.
```
Avoiding non-accessible terms like "see, click, read"
For reference [Accessibility Guidelines for content authors](https://docs.google.com/document/d/1IGuMf16-cwxx8BX1Z9PWsFWsTPxicwTkX_r7UOhjd4E/edit#heading=h.he0f0b5r16e5)

**Review by alaudazzi — APPROVED** [review]
A minor editing suggestion, otherwise LGTM.

**Pull request merged**